src/templates/catalog.html

```diff
@@ -8,38 +8,48 @@
         {% for page_i in range(catalog|length) %}
             {% for thread_i in range(catalog[page_i]['threads']|length) %}
                 {% set thread = catalog[page_i]['threads'][thread_i] %}
-                <div id="{{ thread.no }}" class="thread doc_id_{{ thread.no }}">
+                <div id="{{ thread.no }}" class="thread doc_id_{{ thread.no }}" tabindex="0">
                     <div class="post_data">
-                        <br>
+                       {% if thread.sub %} 
+						<br>
                         <span class="post_title">{{ thread.sub }}</span>
-                        <br>
-                        Posted by <span class="post_author">{{ thread.name }}</span>
+						<br>
+					   {% endif %}
+                        <span class="post_controls">
+                            [<a href="/{{ board }}/thread/{{ thread.no }}" class="btnr parent" target="_blank">View</a>]
+                            [<a href="https://boards.4chan.org/{{ board }}/thread/{{ thread.no }}" class="btnr parent" target="_blank">Original</a>]
+                        </span>
+						
+                        <span class="poster{% if thread.capcode == 'A' %} capcodeAdmin{% endif %}{% if thread.capcode == 'F' %} capcodeFounder{% endif %}{% if thread.capcode == 'M' %} capcodeMod{% endif %}{% if thread.capcode == 'D' %} capcodeDeveloper{% endif %}{% if thread.capcode == 'G' %} capcodeManager{% endif %}{% if thread.capcode == 'V' %} capcodeVerified{% endif %}">Posted by <span class="name post_author{% if thread.capcode == 'A' %} capcodeAdmin{% endif %}{% if thread.capcode == 'F' %} capcodeFounder{% endif %}{% if thread.capcode == 'M' %} capcodeMod{% endif %}{% if thread.capcode == 'D' %} capcodeDeveloper{% endif %}{% if thread.capcode == 'G' %} capcodeManager{% endif %}{% if thread.capcode == 'V' %} capcodeVerified{% endif %}"{% if thread.exif %} title="Exif: {{thread.exif}}"{% endif %}>{{ thread.name }}</span>
```

Comment:
This capcode logic would be better as a macro, e.g. `{% include 'capcode_class.html' %}`. Or, even better, do this mapping on the server using a dict so it's easier to read and modify.